### PR TITLE
Validation cleanup part 1

### PR DESCRIPTION
**Pull request by thockin**
Part 1 of several.  Merge fielderrors and validation together, simplify names.

@lavalamp

### CODE REVIEW

File: pkg/util/fielderrors/fielderrors.go [hunk 1]
```diff
@@ -1,207 +0,0 @@
-/*
```
Comment (lavalamp) [inline pkg/util/fielderrors/fielderrors.go]: It is way too much of a friday for me to review in depth, but I thought these were here to break some cycle? Maybe not. I like where you put it, though I would consider a sub-package there, too.

Comment (thockin) [inline pkg/util/fielderrors/fielderrors.go]: No cycle, and one of the followup PRs does in fact create a sub-pkg :)

No sweat that it is friday - I wanted to get these off my laptop before I broke it.

**Comment by thockin** [comment]
unit test failed for no reason:

```
No changes.	
GitHub pull request #17245 of commit 40cd8851f91c45f90c5780d5516bcf4571f9ad32, no merge conflicts.
Test Result (no failures)
```

**Comment by thockin** [comment]
@k8s-bot unit test this please

@k8s-bot test this please

**Comment by thockin** [comment]
@k8s-bot unit test this please

@k8s-bot test this please

**Comment by thockin** [comment]
@k8s-bot test this please

**Comment by thockin** [comment]
I have rebased this one, but this is going to be a race.  If there's anything I can do to make reviews on this sequence of PRs  easier, please holler

**Comment by thockin** [comment]
I kept these commits distinct on putrpose.  I can do a squash if you want, but I think it is better without

**Comment by wojtek-t** [comment]
@thockin - I'm afraid you need to rebase :(

**Comment by thockin** [comment]
rebased

**Comment by thockin** [comment]
This PR does not even show in the submit-queue.  Weird.  @k8s-merge-robot wtf?

**Comment by thockin** [comment]
@brendandburns @eparis is there somethign about this PR that makes the merge bot not find it, even when searching by PR # or by my name?

**Comment by thockin** [comment]
I see more merges in my future if this doesn't get in soon.  I am considering a manual merge.  @k8s-oncall

I don't know why the bot lost track of this one

**Comment by smarterclayton** [comment]
The bot lost track of another one - found it again this morning

**Comment by wojtek-t** [comment]
Hmm - I've never seen such problems with submit queue.
Why it is still not merged, is because:
1. we have huge problems with scalability tests, and we weren't merging anything for quite a long time
2. merge bot is merging starting from the oldest (with lowest PR number) lgtm-ed PRs, and there are still few older ones...
3. merge-queue is always empty when e2e are red/yellow

Can you please ping me or @lavalamp or @eparis directly once you observe the same behavior?

**Comment by eparis** [comment]
@thockin that is really weird that it just doesn't show up!

**Comment by eparis** [comment]
that was supposed to be @smarterclayton sorry tim!

**Comment by thockin** [comment]
It did show up a while ago and now it is gone again

**Comment by thockin** [comment]
If you search for me, you will find parts 2-6 of this series but not part 1.

**Comment by eparis** [comment]
I'll file an issue on the bot. I just ran a copy of the bot locally and it told me 
```
I1119 12:07:21.435522   28714 github.go:1059] Issue 17245 labels: [cla: yes lgtm size/XXL] isPR: false
```
so for some reason the bot code doesn't think this is a PR and is instead an 'issue'

**Comment by wojtek-t** [comment]
@thockin - this is strange. I can see it in Github e2e tab

**Comment by thockin** [comment]
Can I manually merge?  I see a couple other PRs touching these files and I want to win the race, for once.

**Comment by wojtek-t** [comment]
@thockin - I'm fine with that

**Comment by eparis** [comment]
and I got the github api data (which is wrong/buggy, not my fault!)

**Comment by thockin** [comment]
@eparis: so github is legit getting this wrong?  Or are we using the wrong fields or something?

**Comment by thockin** [comment]
it's back in the submit-queue but blocked on e2e internal

**Comment by eparis** [comment]
looks like github is legit getting it wrong.  At the moment https://api.github.com/repos/kubernetes/kubernetes/issues/17245 is showing a "pull_requests" key. But it appears to be intermittent.

**Comment by eparis** [comment]
https://developer.github.com/v3/issues/#get-a-single-issue

seems to indicate 'pull_request' should always be set (when it is a PR) but that doesn't appear to always be the case....

**Comment by thockin** [comment]
Is there a place to file a bug against github?

**Comment by eparis** [comment]
I e-mailed github support

**Comment by thockin** [comment]
it was on the list and now it is back off the list.  I am repeatedly hitting teh API url looking for failures, the only failure I get is "API rate limit exceeded" :)

**Comment by lavalamp** [comment]
@eparis-- thanks. maybe we can key off of something else in the meantime?

**Comment by thockin** [comment]
This is painful.

How do I run the munger by hand?  I tried:

```
thockin@freakshow:~/src/contrib/mungegithub$ ./mungegithub --dry-run
I1119 13:21:12.410656   21923 github.go:188] Made 0 API calls since the last Reset 0.000000 calls/sec
I1119 13:21:12.806775   21923 mungegithub.go:54] Running mungers
E1119 13:21:12.807053   21923 github.go:63] Ran out of github API tokens. Sleeping for 39.386549198666664 minutes
```

@k8s-oncall, I'm gonna merge it later today if this isn't unstuck

**Comment by eparis** [comment]
you basically MUST use --token-file or --token
```
./mungegithub --token-file=/home/eparis/github-oauth --dry-run --token-file=/home/eparis/github-oauth --v=10 --max-pr-number=17245 --min-pr-number=17245 --period=2m -pr-mungers=submit-queue
```

**Comment by lavalamp** [comment]
This is currently 11th or so in the queue.

**Comment by eparis** [comment]
when we get this far behind I almost want to run 2 merges at the same time. I know it could give us non-linear/obvious e2e failures though  :-(

**Comment by thockin** [comment]
Could we do 2-3 merges per run?

**Comment by eparis** [comment]
We merge as fast as we can run e2e. e2e takes like 20-30 minutes I think. So we only mere 2-3 per hour. It would be possible to run the e2e of 2 PRs at the same time but then you aren't getting good results when the second one merges. If PR A and PR B both passed individually but together they failed we wouldn't detect and e2e failure until PR C.

**Comment by thockin** [comment]
I meant to run e2e on master+A+B and if it passes, merge A and B.  If A+B fails, try A alone.   if we did this 2 or three at a time we could double our throughput with only a little more risk (as long as nobody hand-merges)

**Comment by thockin** [comment]
@lavalamp where do you see the queue?  I still do not see that PR in http://submit-queue.k8s.io/

**Comment by eparis** [comment]
The queue just stopped because google internal e2e just failed: kubernetes-e2e-gke-ci

I understand what you mean now, but I'd have to leave that implementation to someone who can make changes or provide a better/different interface to `Jenkins GCE e2e`

**Comment by eparis** [comment]
The queue is tab 2 called 'Github E2E'   Maybe it should be renamed.

**Comment by lavalamp** [comment]
Yeah, I have so far today filed (or bumped priority to P0) on 5 different flaky test bugs.

**Comment by eparis** [comment]
Might be a real e2e failure.

**Comment by thockin** [comment]
rebased again

**Pull request merged**